eng/scripts/Language-Settings.ps1

```diff
@@ -260,6 +260,260 @@ function Update-java-CIConfig($pkgs, $ciRepo, $locationInDocRepo, $monikerId=$nu
+    $sourcesExtractPath = Join-Path $packageDirectory "sources"
+    Add-Type -AssemblyName System.IO.Compression.FileSystem
+    [System.IO.Compression.ZipFile]::ExtractToDirectory($sourcesJarPath, $sourcesExtractPath)
+
+    if (!(Test-Path "$sourcesExtractPath\com")) {
+      LogWarning "Could not locate 'com' folder extracting $packageArtifact"
+      return $false
+    }
+  }
+  catch 
+  {
+    LogError "Exception while updating checking if package can be documented: $($package.packageGroupId):$($package.packageArtifactId)"
+    LogError $_ 
+    LogError $_.ScriptStackTrace
+    return $false
+  }
+
+  return $true
+}
+
+function PackageDependenciesResolve($package, $workingDirectory) { 
+  $packageDirectory = Join-Path `
+    $workingDirectory `
+    "$($package.packageGroupId)__$($package.packageArtifactId)__$($package.packageVersion)"
+  New-Item -ItemType Directory -Path $packageDirectory -Force
+  
+  $pomArtifactName = "$($package.packageGroupId):$($package.packageArtifactId):$($package.packageVersion):pom"
+  $artifactDownloadOutput = mvn `
+    dependency:copy `
+    -Dartifact="$pomArtifactName" `
+    -DoutputDirectory="$packageDirectory"
+
+  if ($LASTEXITCODE) {
+    LogWarning "Could not download pom artifact: $pomArtifactName"
+    $artifactDownloadOutput | Write-Host
+    return $false
+  }
+
+  # Rename so dependency:copy-dependencies can find the pom file
+  $downloadedPomPath = (Get-ChildItem -File -Path $packageDirectory -Filter '*.pom')[0]
+
+  # -P '!azure-mgmt-sdk-test-jar' excludes the unpublished test jar from
+  # dependencies
+  $copyDependencyOutput = mvn `
+    -f $downloadedPomPath `
+    dependency:copy-dependencies `
+    -P '!azure-mgmt-sdk-test-jar' `
+    -DoutputDirectory="$packageDirectory" 
+  
+  if ($LASTEXITCODE) {
+    LogWarning "Could not resolve dependencies for: $pomArtifactName"
+    $copyDependencyOutput | Write-Host
+    return $false
+  }
+
+  return $true
+}
+
+function ValidatePackage($package, $workingDirectory) { 
+  return (SourcePackageHasComFolder $package $workingDirectory) `
+    -and (PackageDependenciesResolve $package $workingDirectory)
+}
+
+function Update-java-DocsMsPackages($DocsRepoLocation, $DocsMetadata) { 
+  Write-Host "Excluded packages:"
+  foreach ($excludedPackage in $PackageExclusions.Keys) {
+    Write-Host "  $excludedPackage - $($PackageExclusions[$excludedPackage])"
+  }
+
+  $FilteredMetadata = $DocsMetadata.Where({ !($PackageExclusions.ContainsKey($_.Package)) })
+
+  UpdateDocsMsPackages `
+    (Join-Path $DocsRepoLocation 'package.json') `
+    'preview' `
+    $FilteredMetadata
+
+  UpdateDocsMsPackages `
+    (Join-Path $DocsRepoLocation 'package.json') `
+    'latest' `
+    $FilteredMetadata
+}
+
+function UpdateDocsMsPackages($DocConfigFile, $Mode, $DocsMetadata) {
+  $packageConfig = Get-Content $DocConfigFile -Raw | ConvertFrom-Json
+
+  $workingDirectory = Join-Path ([System.IO.Path]::GetTempPath()) ([System.IO.Path]::GetRandomFileName())
```

Comment:
Why do we need a temp directory? If this is to extract and parse the package do we think we can switch to using the package properties instead?

Reply:
Or is this more for checking the dependencies?

Reply:
This is for the purpose of validating that the package will pass the docs CI (dependency checking, having a `com/` folder in the source jar). 

We could easily create a temp directory for every package we validate but that could get difficult to track down when running locally.

Reply:
Ah it isn't about getting simple properties. I agree with you that having one temp directory is better then having a unique one for each package.